core/src/main/java/hudson/util/PluginServletFilter.java

```diff
@@ -142,5 +144,25 @@ public void destroy() {
         list.clear();
     }
 
+    @Restricted(NoExternalUse.class)
+    public static void cleanUp() {
+        PluginServletFilter instance = getInstance(Jenkins.getInstance().servletContext);
+        if (instance != null) {
+            for (Iterator<Filter> iterator = instance.list.iterator(); iterator.hasNext(); ) {
+                Filter f = iterator.next();
+                try {
+                    f.destroy();
+                } catch (RuntimeException e) {
+                    LOGGER.log(Level.WARNING, "Filter " + f + " propagated an exception from its destroy method", e);
+                } catch (Error e) {
+                    throw e; // we are not supposed to catch errors, don't log as could be an OOM
```

Comment:
In fact catching `LinkageError`s is generally wise in a modular application. Not likely in this case.